### PR TITLE
fix: task drag and drop possible again

### PR DESCRIPTION
**Pull request by konstantinschuette**
## Which issues does this pull request close?
closes https://github.com/helpwave/web/issues/726

### CODE REVIEW

File: tasks/components/layout/TasksKanbanBoard.tsx [hunk 1]
```diff
@@ -122,26 +122,30 @@ export const TasksKanbanBoard = ({
...
+    // Item index (order) of the column, which the user drags the task to
     const overIndex = overItems.findIndex(item => item.id !== over?.id)
 
-    if (activeIndex === -1 || overIndex === -1) {
+    // If it can't identify the dragged item
+    if (activeIndex === -1) {
       return
     }
 
+    console.log(overIndex)
```
Comment (MaxSchaefer) [inline tasks/components/layout/TasksKanbanBoard.tsx:135]:
```suggestion
```

**Review by MaxSchaefer — DISMISSED** [review]
Yeah! Its working again. Thank you very much 🎉 Congrats to your first pull-request

File: tasks/components/layout/TasksKanbanBoard.tsx [hunk 2]
```diff
@@ -122,26 +122,29 @@ export const TasksKanbanBoard = ({
...
-    if (activeIndex === -1 || overIndex === -1) {
+    // If it can't identify the dragged item
+    if (activeIndex === -1) {
       return
     }
 
+    // -1 means, that there was no task found, which you dragged your task over
+    const insertAtIndex = overIndex === -1 ? 0 : overIndex
+    // Insert task at insertAtIndex in the overColumn
+    sortedTasks[overColumn].splice(insertAtIndex, 0, sortedTasks[activeColumn][activeIndex] as TaskDTO)
```
Comment (JannikWibker) [inline tasks/components/layout/TasksKanbanBoard.tsx:138]: don't mutate the array here (i.e. use `slice` instead and save a copy of the array)

Comment (MaxSchaefer) [reply tasks/components/layout/TasksKanbanBoard.tsx:138]: @JannikWibker, @konstantinschuette and myself are not quite sure about the benefit. Could you propose a change?

Comment (JannikWibker) [reply tasks/components/layout/TasksKanbanBoard.tsx:138]: [State in react is immutable](https://react.dev/learn/updating-objects-in-state). Doing modifications anyways leads to undefined behaviour / is not supported at all.

```tsx
const newOverColumn = sortedTasks[overColumn].slice(insertAtIndex, 0, sortedTasks[activeColumn][activeIndex] as TaskDTO)

// The last array access is safe as we checked the index before (> -1 and < length; findIndex will not return something >= length)
// Remove dragged task from its origin column, which is the activeColumn
const newActiveColumn = sortedTasks[activeColumn]
  .map((task, i) => i === activeIndex ? { ...task, status: overColumn } : task)
  .filter((item) => item.id !== active.id)

setSortedTasks({
  ...sortedTasks,
  [overColumn]; newOverColumn,
  [activeColumn]: newActiveColumn
})
```

Comment (use-to) [reply tasks/components/layout/TasksKanbanBoard.tsx:138]: Status? Current tasks version is not usable though...

**Review by use-to — APPROVED** [review]
Squasch merge!

**Pull request merged**